corehq/motech/forms.py

```diff
@@ -56,15 +56,38 @@ class BaseConnectionSettingsFormSet(forms.BaseModelFormSet):
 
     def __init__(self, *args, domain, **kwargs):
         super().__init__(*args, **kwargs)
+        self.domain = domain
         self.form_kwargs['domain'] = domain  # Passed by ``FormView.get_form_kwargs()``
 
+    @property
+    def deleted_forms(self):
+        """
+        Returns a list of forms that have been marked for deletion.
+
+        Excludes forms of ConnectionSettings that are in use.
+        """
+        deleted_forms = super().deleted_forms
+        ids_in_use = get_connection_ids_in_use(self.domain)
+        return [f for f in deleted_forms
+                if not (f.is_bound and f.instance.id in ids_in_use)]
```

Comment:
If a ConnectionSettings instance is in use, it will not be deleted. **BUT** currently the user is not informed that their request was ignored.

I am open to any suggestions on the best way to tell the user that some instances were deleted (if multiple instances were selected) but others were not because they are in use.

Reply:
`messages` is an easy way, and better than nothing

Reply:
In the end I used normal validation to block deletions, and the messaging that comes with that. 19a08c76cad